retrofit/src/main/java/retrofit/ExecutorCallAdapterFactory.java

```diff
@@ -47,14 +47,36 @@
   static final class ExecutorCallbackCall<T> implements Call<T> {
     private final Executor callbackExecutor;
     private final Call<T> delegate;
+    private volatile boolean canceled;
 
     ExecutorCallbackCall(Executor callbackExecutor, Call<T> delegate) {
       this.callbackExecutor = callbackExecutor;
       this.delegate = delegate;
     }
 
-    @Override public void enqueue(Callback<T> callback) {
-      delegate.enqueue(new ExecutorCallback<>(callbackExecutor, callback));
+    @Override public void enqueue(final Callback<T> callback) {
+      delegate.enqueue(new Callback<T>() {
+        @Override public void onResponse(final Response<T> response, final Retrofit retrofit) {
+          callbackExecutor.execute(new Runnable() {
+            @Override public void run() {
+              if (canceled) {
+                // Emulate OkHttp's behavior of throwing/delivering an IOException on cancelation
```

Comment:
This change is for handling the case where the `Call` is cancelled after the response has been delivered by OkHttp, while still being delivered by the callback `Executor`. As mentioned in #1135, this would cause the result to be delivered to the callback even after cancellation.